libs/utils/utils.js

```diff
@@ -1158,8 +1173,10 @@ async function processSection(section, config, isDoc) {
   }
 
   if (section.preloadLinks.length) {
-    const preloads = section.preloadLinks.map((block) => loadBlock(block));
+    const [modals, nonModals] = partition(section.preloadLinks, (block) => block.classList.contains('modal'));
+    const preloads = nonModals.map((block) => loadBlock(block));
     await Promise.all(preloads);
+    modals.map((block) => loadBlock(block));
```

Comment:
**Nits:**

1. I think this particular part is more complicated than necessary. I find the partition function hard to read.
2. I think marquee block preload should not be mixed in the same PR as whatever we're trying to do here with modals. These are different problem spaces.

Reply:
Since the arrays are very short and the `fn` is simple, `partition` could use `filter`.  This does make it so the array is parsed twice, but there should be a negligible change in js perf.

```
function partitionArray(arr, fn) {
  const matchedItems = arr.filter(fn);
  const unmatchedItems = arr.filter(item => !fn(item));
  return [matchedItems, unmatchedItems];
}
```

Reply:
Thank you for your suggestion. The current function works as expected and there are tests to guard any changes in the future. It's a single purpose function which unlikely needs any more modification. I will leave it as is.

Reply:
non-blocking nit: semantically maybe a forEach makes more sense than map for modals? It could save a tiny bit of memory too as a mapping array won't be created, but really negligible even if the engine doesn't optimize it away